src/visualization/types/Graph/view.tsx

```diff
@@ -223,23 +195,55 @@ const XYPlot: FC<Props> = ({properties, result, timeRange, annotations}) => {
     ],
   }
 
-  if (isFlagEnabled('annotations') && annotations) {
+  if (isFlagEnabled('annotations') && annotationsModeIsActive) {
+    const createAnnotation = userModifiedAnnotation => {
+      const {message, startTime} = userModifiedAnnotation
+      dispatch(
+        writeThenFetchAndSetAnnotations([
+          {
+            summary: message,
+            startTime: new Date(startTime).getTime(),
+            endTime: new Date(startTime).getTime(),
+          },
+        ])
+      )
+    }
+
+    const doubleClickHandler = (
+      plotInteraction: InteractionHandlerArguments
+    ) => {
+      dispatch(
+        showOverlay(
+          'add-annotation',
+          {
+            createAnnotation,
+            startTime: plotInteraction.valueX,
+          },
+          dismissOverlay
+        )
+      )
+    }
+
+    config.interactionHandlers = {
+      doubleClick: doubleClickHandler,
+    }
+
     // everything is under the 'default' category for now:
-    const actualAnnotations: any[] = annotations.default ?? null
+    const selectedAnnotations: any[] = annotations?.default ?? []
 
-    if (actualAnnotations && actualAnnotations.length) {
+    if (selectedAnnotations.length) {
```

Comment:
since it defaults to an empty array, we can just check the length here, rather than existence *and* length.